lib/puppet/provider/network_config/redhat.rb

```diff
@@ -146,26 +142,19 @@ def self.munge(pairs)
     # For each interface attribute that we recognize it, add the value to the
     # hash with our expected label
     NAME_MAPPINGS.each_pair do |type_name, redhat_name|
-      if (val = pairs[redhat_name])
-        # We've recognized a value that maps to an actual type property, delete
-        # it from the pairs and copy it as an actual property
-        pairs.delete(redhat_name)
-        props[type_name] = val
-      end
+      next unless (val = pairs[redhat_name])
+      pairs.delete(redhat_name)
+      props[type_name] = val
     end
 
     # if we encounter VLAN=yes set the interface mode to :vlan
     pairs.each_pair do |key, val|
-      if (key == 'VLAN' and val == 'yes')
-        props[:mode] = :vlan
-      end
+      props[:mode] = :vlan if key == 'VLAN' && val == 'yes'
     end
     pairs.delete('VLAN')
 
     # mode is a property so it should always have a value
-    unless (props.key?(:mode))
-      props[:mode] = :raw 
-    end
+    props[:mode] = :raw unless props.key?(:mode)
```

Comment:
Looks like another candidate for ||= instead of the unless as it says it should always have a value and looks like that value shouldn't be able to be nil in this case...